### DISCUSSION TITLE
Deno Language Server client: couldn't create connection to server.

### DISCUSSION THREAD

**Discussion by mikoloism** [denoland/vscode_deno#1188] [Q&A] [answered]
**Describe the bug**

> VSCode Output of `Deno Language Server`
>
> Launching server using command /${userHome}/.deno/bin/deno failed. Error: spawn /${userHome}/.deno/bin/deno ENOENT

**Versions**

vscode: `1.85.0` deno: `1.38.5` extension: `v3.29.0`

**Comment by bartlomieju** [comment]
If you do `which deno` what output do you get?

**Reply by Meherab003** [reply]
/c/Users/meherab/.deno/bin/deno

**Comment by mikoloism** [comment]
It is in "$HOME/.deno/bin/deno" and I configured this path to ".vscode/settings.json".

Is there a way to use Deno without config the 'deno.path' property inside '.vscode/settings.json'?

**Comment by bartlomieju** [comment, marked as answer]
It should already work this way - are you sure the VSCode is resolving `${userHome}` to the right directory?

**Comment by mikoloism** [comment]
> It should already work this way

Yes should Work but not!

>  are you sure the VSCode is resolving `${userHome}` to the right directory?

How can i testing this?

**Comment by Strajk** [comment]
I had deno installed via homebrew and was getting this error. 
Got so frustrated trying to fix it that I uninstalled the brew instance, installed it via `curl -fsSL https://deno.land/x/install/install.sh | sh`, restarted VSCode and now it works ¯\_(ツ)_/¯

Maybe it will help someone Googling this error.

**Comment by ilmatic** [comment]
> I had deno installed via homebrew and was getting this error. Got so frustrated trying to fix it that I uninstalled the brew instance, installed it via `curl -fsSL https://deno.land/x/install/install.sh | sh`, restarted VSCode and now it works ¯_(ツ)_/¯
> 
> Maybe it will help someone Googling this error.

Helped me! If it's not an easy fix might be worth sticking a note in the extension description that it doesn't work with homebrew installs.

**Comment by guizmo** [comment]
> I had deno installed via homebrew and was getting this error. Got so frustrated trying to fix it that I uninstalled the brew instance, installed it via `curl -fsSL https://deno.land/x/install/install.sh | sh`, restarted VSCode and now it works ¯_(ツ)_/¯
> 
> Maybe it will help someone Googling this error.

Perfect  !!!! Thanks
```
brew uninstall deno
curl -fsSL https://deno.land/x/install/install.sh | sh
```

**Comment by RepComm** [comment]
Re-installing deno on ubuntu using similar instructions as mentioned already also solved this for me.

**Comment by riderx** [comment]
Can we just get the extension accept brew install that doesn't make sense

**Comment by zing-rsa** [comment]
In my case, I hadn't installed deno with homebrew, but running `curl -fsSL https://deno.land/x/install/install.sh | sh` resolved the issue anyway.

**Comment by joaovitorbf** [comment]
This error also happens when Deno is installed via deno-cli through NPM on Windows (this is not an official way of installing Deno, but still, it seems like it doesn't like other types of installations, even if the Deno binary is in Path or added to the Deno: Path vscode config variable).

Installing Deno using NPM deno-cli is sometimes useful if you are behind policies that require you to install packages only through a registry proxy, as the proxy validates the deno-cli installation package and allows you to use it.

**EDIT: I was able to make it work by using the path "C:\Users\USERNAME\AppData\Roaming\npm\node_modules\deno-bin\bin\deno.exe" instead of "C:\Users\USERNAME\AppData\Roaming\npm\deno".**

**Comment by nayeemrmn** [comment]
When this occurs it's definitely a user env issue. Let's convert this to a discussion for help.
